Firestore/core/src/firebase/firestore/core/firestore_client.h

```diff
@@ -0,0 +1,199 @@
+   */
+  void Shutdown(util::StatusCallback callback);
+
+  /**
+   * Passes a callback that is triggered when all the pending writes at the
+   * time when this method is called received server acknowledgement.
+   * An acknowledgement can be either acceptance or rejections.
+   */
+  void WaitForPendingWrites(util::StatusCallback callback);
+
+  /** Disables the network connection. Pending operations will not complete. */
+  void DisableNetwork(util::StatusCallback callback);
+
+  /** Enables the network connection and requeues all pending operations. */
+  void EnableNetwork(util::StatusCallback callback);
+
+  /** Starts listening to a query. */
+  std::shared_ptr<QueryListener> ListenToQuery(
+      Query query,
+      ListenOptions options,
+      ViewSnapshot::SharedListener&& listener);
+
+  /** Stops listening to a query previously listened to. */
+  void RemoveListener(const std::shared_ptr<core::QueryListener>& listener);
+
+  /**
+   * Retrieves a document from the cache via the indicated callback. If the doc
+   * doesn't exist, an error will be sent to the callback.
+   */
+  void GetDocumentFromLocalCache(const api::DocumentReference& doc,
+                                 api::DocumentSnapshot::Listener&& callback);
+
+  /**
+   * Retrieves a (possibly empty) set of documents from the cache via the
+   * indicated callback.
+   */
+  void GetDocumentsFromLocalCache(const api::Query& query,
+                                  api::QuerySnapshot::Listener&& callback);
+
+  /**
+   * Write mutations. callback will be notified when it's written to the
+   * backend.
+   */
+  void WriteMutations(std::vector<model::Mutation>&& mutations,
+                      util::StatusCallback callback);
+
+  /**
+   * Tries to execute the transaction in update_callback up to retries times.
+   */
+  void Transaction(int retries,
+                   TransactionUpdateCallback update_callback,
+                   TransactionResultCallback result_callback);
+
+  /** The database ID of the DatabaseInfo this client was initialized with. */
+  const model::DatabaseId& database_id() const {
+    return database_info_.database_id();
+  }
+
+  /**
+   * Dispatch queue for user callbacks / events. This will often be the "Main
+   * Dispatch Queue" of the app but the developer can configure it to a
+   * different queue if they so choose.
+   */
+  const std::shared_ptr<util::Executor>& user_executor() const {
+    return user_executor_;
+  }
+
+  /** For usage in this class and testing only. */
+  const std::shared_ptr<util::AsyncQueue>& worker_queue() const {
+    return worker_queue_;
+  }
+  bool is_shutdown() const;
+
+ private:
+  FirestoreClient(
+      const DatabaseInfo& database_info,
+      std::shared_ptr<auth::CredentialsProvider> credentials_provider,
+      std::shared_ptr<util::Executor> user_executor,
+      std::shared_ptr<util::AsyncQueue> worker_queue);
+  /**
+   * Initializes this instance of the client. This should be called right after
```

Comment:
Nit: now that this function is private, perhaps use the regular `//` comments instead of Javadoc-ey `/** */`?

Reply:
Done.